core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1143,7 +1143,8 @@ class KafkaController(val config: KafkaConfig, zkUtils: ZkUtils, time: Time, met
 
     override def process(): Unit = {
       if (!isActive) return
-      val curBrokers = currentBrokerList.map(_.toInt).toSet.flatMap(zkUtils.getBrokerInfo)
+      // Double read to increase the odds of processing recent broker changes in a single ControllerEvent (KAFKA-5502).
```

Comment:
How about "Read the current broker list from ZK again instead of using currentBrokerList to increase the odds of processing recent broker changes in a single ControllerEvent (KAFKA-5502)."?